# Actividad Evaluable: Obtención de estadisticas descriptivas 

Importando las librerías necesarias para procesar los datos de Spotify y cargando datos (como el .csv esta en el mismo directorio que el código no tengo que poner toda la ruta) (W3Schools.com, s. f.):

In [2]:
import pandas as pd

df = pd.read_csv("spotify.csv")

Verifico la cantidad de datos (no nulos, osea, con valor), las variables de cada vector (en este caso columnas) y que tipo de dato es cada columna (el Dtype) con el método info() (Encontrar el Tamaño de un Marco de Datos En Python, 16d. C.),

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 114000 entries, 0 to 113999
Data columns (total 21 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   Unnamed: 0        114000 non-null  int64  
 1   track_id          114000 non-null  object 
 2   artists           113999 non-null  object 
 3   album_name        113999 non-null  object 
 4   track_name        113999 non-null  object 
 5   popularity        114000 non-null  int64  
 6   duration_ms       114000 non-null  int64  
 7   explicit          114000 non-null  bool   
 8   danceability      114000 non-null  float64
 9   energy            114000 non-null  float64
 10  key               114000 non-null  int64  
 11  loudness          114000 non-null  float64
 12  mode              114000 non-null  int64  
 13  speechiness       114000 non-null  float64
 14  acousticness      114000 non-null  float64
 15  instrumentalness  114000 non-null  float64
 16  liveness          11

Para saber más de todos mis datos utilizaré el método *describe()* junto con el parametro *include = "all"*,  para generar un análisis de todas las estadísticas sin importar si son datos númericos o categoricos (datos que se pueden clasificar pero no tienen una jerarquía u orden matemático (Aprendizaje Automático de Datos Categóricos Con el Tutorial de Python, 2024)) y así llegar a conclusiones a partir de este informe de todas las columnas (GeeksforGeeks, 2025) y también determinaré la mediana de mis columnas númericas con el método *median()* y su parametro *numeric_only* como true, ya que para determinar una mediana el dato debe ser númerico (Malli, 2024).

In [4]:
print(df.describe(include="all"))

           Unnamed: 0                track_id      artists  \
count   114000.000000                  114000       113999   
unique            NaN                   89741        31437   
top               NaN  6S3JlDAGk3uu3NtZbPnuhS  The Beatles   
freq              NaN                       9          279   
mean     56999.500000                     NaN          NaN   
std      32909.109681                     NaN          NaN   
min          0.000000                     NaN          NaN   
25%      28499.750000                     NaN          NaN   
50%      56999.500000                     NaN          NaN   
75%      85499.250000                     NaN          NaN   
max     113999.000000                     NaN          NaN   

                        album_name       track_name     popularity  \
count                       113999           113999  114000.000000   
unique                       46589            73608            NaN   
top     Alternative Christmas 2022  Run Rudol

In [5]:
print(df.median(numeric_only=True))

Unnamed: 0           56999.500000
popularity              35.000000
duration_ms         212906.000000
explicit                 0.000000
danceability             0.580000
energy                   0.685000
key                      5.000000
loudness                -7.004000
mode                     1.000000
speechiness              0.048900
acousticness             0.169000
instrumentalness         0.000042
liveness                 0.132000
valence                  0.464000
tempo                  122.017000
time_signature           4.000000
dtype: float64


Para terminar, puedo decir que de las 114000 canciones, me sorprende que haya 9 canciones con un mismo id, siendo que, por convención, un id debe ser único, aunque no necesariamente ya que puede ser una misma canción pero en diferente álbum, lo cuál hace sentido, ya que sino, tendríamos en Spotify o cualquier app de música la misma canción en diferentes álbumes con diferentes cantidades de escuchas, lo cúal no hace sentido; por otro lado no me sorprende ver a los Beatles como el artista más frecuente de los 31437 del csv dada la influencia de ellos y lo clave que han sido para la historia de la música; también me sorprende como un álbum de navidad es el nombre de álbum más común para 195 canciones, lo que implica que ese álbum tiene 195 canciones de "navidad alternativa".

Por otro lado, también es interesante como el nombre de canción "Run Rudolph Run" es el más presente, probablemente como Cover o canción más presente en diferentes álbumes; también es interesante como en promedio las canciones son un tercio de populares, y hay una gran dispersión de la popularidad de las mismas, lo que significa que pocas canciones de las 114000 son muy populares al momento de la recolección de los datos.

En cuento a la duración de las canciones no me sorprende que sean de 3.8 minutos ya que es el estándar de duración de canciones número uno de todos los tiempos en EUA (Thrasher, 2023) y  que la signatura sea el clasico de 4/4, con la excepción de 4/5; similar caso con la "bailabilidad", energía, tempo, y si es explicita o no la canción ya que la mayoría de personas escucha canciones en fiestas, reuniones, eventos, y situciones alegres, es decir, con un tempo alto, "movido", con energía que tienda a ser alta y también que tiendan a ser bailables las canciones, y dado a las situaciones en que es más común escuchar música, también se tiene a que no se hagan todas explicitas, inclusive hay artistas que hacen versiones no explicitas para que suenen más sus canciones.

Más alla de esos factores, lo que si más me sorprende es que en promedio de las 114000 canciones a estas se les reduce el volumen como 8 decibelios y por ende no tienen tanta valencia acústica, se consideran no tan vocales, no son tan acústicas, ni instrumentales o vivas pero el género más popular de los 114 géneros es el género acústico.

Referencias:

1. W3Schools.com. (s. f.). https://www.w3schools.com/python/pandas/pandas_csv.asp

2. Encontrar el tamaño de un marco de datos en Python. (2025, 16 enero). Datacamp. Recuperado 28 de octubre de 2025, de https://www.datacamp.com/es/tutorial/python-dataframe-size

3. GeeksforGeeks. (2025, 26 julio). Pandas DataFrame describe() Method. GeeksforGeeks. https://www.geeksforgeeks.org/pandas/python-pandas-dataframe-describe-method/

4. Aprendizaje automático de datos categóricos con el tutorial de Python. (2024, 3 mayo). Datacamp. Recuperado 28 de octubre de 2025, de https://www.datacamp.com/es/tutorial/categorical-data

5. Thrasher, D. (2023, 15 mayo). How Long Should a Song Be? [Radio, Streaming, and Social!] - Improve Songwriting. Improve Songwriting. https://improvesongwriting.com/how-long-should-a-song-be/

6. Malli. (2024, 24 julio). Pandas DataFrame median() Method. Spark By {Examples}. https://sparkbyexamples.com/pandas/pandas-dataframe-median-method/